src/Specs.ts

```diff
@@ -87,13 +87,13 @@ export class Specs {
 			throw APPEND_SPEC_ERROR;
 		}
 
-		// WS
+		// WS OR HTTP
 		this._specs.appendSpec(
 			MODULES.SUBSTRATE,
-			this._specs.getSpec(CONFIG.WS_URL, 'Websocket URL', {
+			this._specs.getSpec(CONFIG.URL, 'Websocket or HTTP URL', {
 				default: 'ws://127.0.0.1:9944',
 				mandatory: true,
-				regexp: /^wss?:\/\/.*(:\d{4,5})?$/,
+				regexp: /^(ws|wss|http)?:\/\/.*(:\d{4,5})?$/,
```

Comment:
```suggestion
				regexp: /^(ws|wss|http)?:\/\/.*$/,
```

Just for sake of clarity. This optional part of regexp `(:\d{4,5})?` totally have no meaning after `.*`

Reply:
good catch, thanks.

@TarikGul is it possible to get rid of this regex and use a URI/URL parser instead?

Reply:
@sergejparity Thanks for that catch.

@niklasad1 I think it's probably best to leave the regex since the support is built in directly into `confmgr`, it also gives some solid logging associated with it, and removing it and doing our own check would complicate the SpecConfig. 

If it had an option to accept a callback to do custom validation I would be down, but it currently doesn't. Heres a link to the options available: https://gitlab.com/chevdor/confmgr/-/blob/master/src/types/optionTypes.ts#L9-15